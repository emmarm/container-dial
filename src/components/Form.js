import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const StyledForm = styled('form')({
  display: 'flex',
  flexDirection: 'column',
  width: '100%'
});

const Form = ({ children, onSubmit }) => (
  <StyledForm onSubmit={onSubmit}>{children}</StyledForm>
);

Form.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  onSubmit: PropTypes.func
};

export default Form;
