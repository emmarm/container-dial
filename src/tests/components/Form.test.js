import React from 'react';
import { shallow } from 'enzyme';

import Form from '../../components/Form';

describe('Form component', () => {
  const wrapper = shallow(<Form />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a Form form', () => {
    const form = wrapper.find('Styled(form)');
    expect(form.length).toBe(1);
  });
});
