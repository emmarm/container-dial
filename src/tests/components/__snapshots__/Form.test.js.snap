// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches snapshot 1`] = `<Styled(form) />`;
